### PR TITLE
Test spaceless ids (refactor the schema validation tests)

### PR DESCRIPTION
**Pull request by jgreben**
This PR was originally about testing the "spaceless" IDs in the JSON schema (See https://github.com/LD4P/sinopia/pull/201) but morphed into refatoring the schema validation tests to happen at the component/unit-test level in stead of integration tests (which were incomplete and error-prone). There is still one integration test for the malformed JSON test which was dificult to test at the unit-test level. A beneficial side-effect is that this refactoring increases overall code-coverage.

### CODE REVIEW

File: __tests__/components/RootComponent.test.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import 'jsdom-global/register'
```
Comment (ndushay) [inline __tests__/components/RootComponent.test.js]: is this new import needed?

File: __tests__/components/editor/ImportResourceTemplate.test.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // Copyright 2019 Stanford University see Apache2.txt for license
 
+import 'jsdom-global/register'
```
Comment (ndushay) [inline __tests__/components/editor/ImportResourceTemplate.test.js]: ditto about this import statement ...

File: __tests__/integration/schemaValidation.test.js [hunk 3]
```diff
@@ -2,145 +2,23 @@
...
-    })
-
-    describe('not schema valid', () => {
-      let dialog
-      beforeAll(async() => {
-        await page.goto('http://127.0.0.1:8888/import')
-        await page.waitForSelector('button#ImportProfile')
-        await page.click('button#ImportProfile')
-        await pupExpect(page).toMatch('select a file to upload:')
+  it('Displays and error message', async () => {
```
Comment (ndushay) [inline __tests__/integration/schemaValidation.test.js]: typo ?  "an" for "and"?

File: __tests__/components/RootComponent.test.js (hunk 1 above)
Comment (jgreben) [reply __tests__/components/RootComponent.test.js]: Yes, actually. This has to do with the refactoring of ImportFileZone and moving the scope of `new window.FileReader()` to the component level. When the tests all run together apparently we need a test DOM from the top down RootComponent => (App) => ImportResourceTemplate => ImportFileZone in order for the test to recognize the existence of the `window` object.

File: __tests__/components/editor/ImportResourceTemplate.test.js (hunk 2 above)
Comment (jgreben) [reply __tests__/components/editor/ImportResourceTemplate.test.js]: ditto above comment..

File: __tests__/integration/schemaValidation.test.js (hunk 3 above)
Comment (jgreben) [reply __tests__/integration/schemaValidation.test.js]: Doh...

**Pull request merged**